presto-common/src/main/java/com/facebook/presto/common/type/Uuids.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.type;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class Uuids
+{
+    private Uuids() {}
+
+    public static long[] rawUuidValuesFromBigEndian(byte[] bytes)
+    {
+        long[] result = new long[2];
+        rawUuidValuesFromBigEndian(result, bytes, 0);
+        return result;
+    }
+
+    public static void rawUuidValuesFromBigEndian(long[] buffer, byte[] bytes, int dstIndex)
```

Comment:
why not return the bytes?

Reply:
This is to prevent memory allocations so that a single buffer may be re-used for multiple invocations of this method.